### PR TITLE
reinstate reactive mariadb tests using r2dbc-mysql

### PR DESCRIPTION
**Pull request by mshima**
- switch mariadb to use r2dbc-mysql (again) commit https://github.com/jhipster/generator-jhipster/pull/24199/commits/a768dd3c92ae99f0407ed7decce117c0cb8a40c9
- add prodDatabaseDriver to application

Fixes https://github.com/jhipster/generator-jhipster/issues/20363.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/spring-data-relational/internal/dependencies.mts [hunk 1]
```diff
@@ -31,6 +31,32 @@ const testcontainerFileForDB = {
   postgresql: 'PostgreSqlTestContainer.java',
 };
 
+type JavaArtifact = { groupId: string; artifactId: string };
+export type DatabaseArtifact = { jdbc: JavaArtifact; r2dbc: JavaArtifact };
+
+const databaseArchifactForDB: Record<string, DatabaseArtifact> = {
```
Comment (DanielFran) [inline generators/spring-data-relational/internal/dependencies.mts:37]:
```suggestion
const databaseArtifactForDB: Record<string, DatabaseArtifact> = {
```

File: generators/spring-data-relational/internal/dependencies.mts [hunk 2]
```diff
@@ -31,6 +31,32 @@ const testcontainerFileForDB = {
...
+    jdbc: { groupId: 'com.mysql', artifactId: 'mysql-connector-j' },
+    r2dbc: { groupId: 'io.asyncer', artifactId: 'r2dbc-mysql' },
+  },
+  postgresql: {
+    jdbc: { groupId: 'org.postgresql', artifactId: 'postgresql' },
+    r2dbc: { groupId: 'org.postgresql', artifactId: 'r2dbc-postgresql' },
+  },
+};
+
+export const getDatabaseDriverForDatabase = (databaseType: string) => databaseArchifactForDB[databaseType];
```
Comment (DanielFran) [inline generators/spring-data-relational/internal/dependencies.mts:58]:
```suggestion
export const getDatabaseDriverForDatabase = (databaseType: string) => databaseArtifactForDB[databaseType];
```

File: generators/spring-data-relational/internal/dependencies.mts [hunk 3]
```diff
@@ -121,34 +147,34 @@ export const getDatabaseTypeMavenDefinition: (
     mariadb: {
       jdbc: {
         dependencies: [
-          { inProfile, groupId: 'org.mariadb.jdbc', artifactId: 'mariadb-java-client' },
+          { inProfile, ...databaseArchifactForDB.mariadb.jdbc },
```
Comment (DanielFran) [inline generators/spring-data-relational/internal/dependencies.mts:150]:
```suggestion
          { inProfile, ...databaseArtifactForDB.mariadb.jdbc },
```

File: generators/spring-data-relational/internal/dependencies.mts [hunk 4]
```diff
@@ -121,34 +147,34 @@ export const getDatabaseTypeMavenDefinition: (
...
           { groupId: 'org.testcontainers', artifactId: 'mariadb', scope: 'test' },
         ],
       },
       r2dbc: {
-        dependencies: [{ inProfile, groupId: 'org.mariadb', artifactId: 'r2dbc-mariadb' }],
+        dependencies: [{ inProfile, ...databaseArchifactForDB.mariadb.r2dbc }],
```
Comment (DanielFran) [inline generators/spring-data-relational/internal/dependencies.mts:155]:
```suggestion
        dependencies: [{ inProfile, ...databaseArtifactForDB.mariadb.r2dbc }],
```

File: generators/spring-data-relational/internal/dependencies.mts [hunk 5]
```diff
@@ -121,34 +147,34 @@ export const getDatabaseTypeMavenDefinition: (
...
       },
     },
     mssql: {
       jdbc: {
         dependencies: [
-          { inProfile, groupId: 'com.microsoft.sqlserver', artifactId: 'mssql-jdbc' },
+          { inProfile, ...databaseArchifactForDB.mssql.jdbc },
```
Comment (DanielFran) [inline generators/spring-data-relational/internal/dependencies.mts:161]:
```suggestion
          { inProfile, ...databaseArtifactForDB.mssql.jdbc },
```

File: generators/spring-data-relational/internal/dependencies.mts [hunk 6]
```diff
@@ -121,34 +147,34 @@ export const getDatabaseTypeMavenDefinition: (
...
           { groupId: 'org.testcontainers', artifactId: 'mssqlserver', scope: 'test' },
         ],
       },
       r2dbc: {
-        dependencies: [{ inProfile, groupId: 'io.r2dbc', artifactId: 'r2dbc-mssql' }],
+        dependencies: [{ inProfile, ...databaseArchifactForDB.mssql.r2dbc }],
```
Comment (DanielFran) [inline generators/spring-data-relational/internal/dependencies.mts:166]:
```suggestion
        dependencies: [{ inProfile, ...databaseArtifactForDB.mssql.r2dbc }],
```

File: generators/spring-data-relational/internal/dependencies.mts [hunk 7]
```diff
@@ -121,34 +147,34 @@ export const getDatabaseTypeMavenDefinition: (
...
       },
     },
     mysql: {
       jdbc: {
         dependencies: [
-          { inProfile, groupId: 'com.mysql', artifactId: 'mysql-connector-j' },
+          { inProfile, ...databaseArchifactForDB.mysql.jdbc },
```
Comment (DanielFran) [inline generators/spring-data-relational/internal/dependencies.mts:172]:
```suggestion
          { inProfile, ...databaseArtifactForDB.mysql.jdbc },
```

File: generators/spring-data-relational/internal/dependencies.mts [hunk 8]
```diff
@@ -121,34 +147,34 @@ export const getDatabaseTypeMavenDefinition: (
...
           { groupId: 'org.testcontainers', artifactId: 'mysql', scope: 'test' },
         ],
       },
       r2dbc: {
-        dependencies: [{ inProfile, groupId: 'io.asyncer', artifactId: 'r2dbc-mysql' }],
+        dependencies: [{ inProfile, ...databaseArchifactForDB.mysql.r2dbc }],
```
Comment (DanielFran) [inline generators/spring-data-relational/internal/dependencies.mts:177]:
```suggestion
        dependencies: [{ inProfile, ...databaseArtifactForDB.mysql.r2dbc }],
```

File: generators/spring-data-relational/internal/dependencies.mts [hunk 9]
```diff
@@ -163,12 +189,12 @@ export const getDatabaseTypeMavenDefinition: (
     postgresql: {
       jdbc: {
         dependencies: [
-          { inProfile, groupId: 'org.postgresql', artifactId: 'postgresql' },
+          { inProfile, ...databaseArchifactForDB.postgresql.jdbc },
```
Comment (DanielFran) [inline generators/spring-data-relational/internal/dependencies.mts:192]:
```suggestion
          { inProfile, ...databaseArtifactForDB.postgresql.jdbc },
```

File: generators/spring-data-relational/internal/dependencies.mts [hunk 10]
```diff
@@ -163,12 +189,12 @@ export const getDatabaseTypeMavenDefinition: (
...
           { groupId: 'org.testcontainers', artifactId: 'postgresql', scope: 'test' },
         ],
       },
       r2dbc: {
-        dependencies: [{ inProfile, groupId: 'org.postgresql', artifactId: 'r2dbc-postgresql' }],
+        dependencies: [{ inProfile, ...databaseArchifactForDB.postgresql.r2dbc }],
```
Comment (DanielFran) [inline generators/spring-data-relational/internal/dependencies.mts:197]:
```suggestion
        dependencies: [{ inProfile, ...databaseArtifactForDB.postgresql.r2dbc }],
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

**Comment by mshima** [comment]
Thanks @DanielFran

**Review by DanielFran — APPROVED** [review]

**Pull request merged**